src/middleware.js

```diff
@@ -16,17 +20,30 @@ const generateEditPageURL = (handle) => `${SITE_URL}/tickets/${handle}/edit`;
 export async function middleware(req) {
   try {
     const { pathname } = req.nextUrl;
-
-    // Exclude static files and API routes
-    if (
-      pathname.startsWith('/_next/') ||
-      pathname.startsWith('/favicon.ico') ||
-      pathname.startsWith('/api/')
-    ) {
-      return NextResponse.next();
+    
+    try {
+      const isLoggedIn = await checkCookie('neon_login_indicator');
+      if (pathname === '/' && isLoggedIn) {
+        try {
+          const referer = await getReferer();
+          if (
+            referer.includes(process.env.VERCEL_BRANCH_URL) ||
+            referer.includes(process.env.NEXT_PUBLIC_DEFAULT_SITE_URL)
+          ) {
+            return NextResponse.redirect(new URL(`${SITE_URL}/home`));
+          }
+        } catch (error) {
+          console.error('Error getting referer:', error);
+        }
+        return NextResponse.redirect(new URL(LINKS.console));
```

Comment:
AFAIK `redirect` accepts a `string` as well as a `URL`, and since `LINKS.console` is a full URL string, this should simply be enough:

```suggestion
        return NextResponse.redirect(LINKS.console);
```